dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/DatadogCore.kt

```diff
@@ -173,20 +177,21 @@ internal class DatadogCore(
     }
 
     /** @inheritDoc */
+    @AnyThread
     override fun addUserProperties(extraInfo: Map<String, Any?>) {
         coreFeature.userInfoProvider.addUserProperties(extraInfo)
     }
 
     /** @inheritDoc */
+    @AnyThread
     override fun clearAllData() {
         features.values.forEach {
             it.clearAllData()
         }
-        // TODO RUM-1462 address Thread safety
-        @Suppress("ThreadSafety") // removal of the data is done in synchronous manner
-        coreFeature.deleteLastViewEvent()
-        @Suppress("ThreadSafety") // removal of the data is done in synchronous manner
-        coreFeature.deleteLastFatalAnrSent()
+        getPersistenceExecutorService().submitSafe("Clear all data", internalLogger) {
```

Comment:
isn't this going to create issues if we expect this method to return in the same stack immediately ?

Reply:
There's no such expectations, and it would mostly create issues if called from the main thread on a running app.